### PR TITLE
Add more data_lake operations

### PR DESCRIPTION
**Pull request by thovoll**
#494

### CODE REVIEW

File: sdk/core/src/request_options/content_length.rs [hunk 1]
```diff
@@ -5,7 +5,6 @@ use http::request::Builder;
 pub struct ContentLength(i32);
 
 impl ContentLength {
-    /// Create a new `ContentLength`
```
Comment (rylev) [inline sdk/core/src/request_options/content_length.rs:8]: Why remove this (other than it being somewhat obvious)?

File: sdk/core/src/request_options/rename_source.rs [hunk 2]
```diff
@@ -0,0 +1,42 @@
+use crate::headers::*;
+use crate::AddAsHeader;
+use http::request::Builder;
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct RenameSource<'a>(&'a str);
```
Comment (rylev) [inline sdk/core/src/request_options/rename_source.rs:6]: Is this used anywhere but in storage?

Also, it would be nice to have a short doc comment description of what this is

```
An optional file or directory to be renamed.
```

File: sdk/storage/examples/data_lake_00.rs [hunk 3]
```diff
@@ -63,44 +62,16 @@ async fn main() -> Result<(), Box<dyn Error + Send + Sync>> {
...
     println!("getting file system properties...");
-    let get_fs_props_response = file_system.get_properties().execute().await?;
+    let get_fs_props_response = file_system_client.get_properties().execute().await?;
     println!(
         "get file system properties response == {:?}",
         get_fs_props_response
     );
     println!();
 
-    let file_name = "example-file.txt";
```
Comment (rylev) [inline sdk/storage/examples/data_lake_00.rs:73]: Why is this removed?

File: sdk/storage/examples/data_lake_01.rs [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+        .append_to_file(
+            Context::default(),
+            file_path,
+            bytes,
+            0,
+            FileAppendOptions::default(),
+        )
+        .await?;
+    println!("append to file response == {:?}", append_to_file_response);
+    println!();
```
Comment (rylev) [inline sdk/storage/examples/data_lake_01.rs:82]: nit: it might be cleaner to just have a `\n` at the end of the previous `println!`

File: sdk/storage/tests/data_lake.rs [hunk 5]
```diff
@@ -122,3 +105,89 @@ async fn test_data_lake_file_system_functions() -> Result<(), Box<dyn Error + Se
 
     Ok(())
 }
+
+#[tokio::test]
+async fn test_data_lake_file_functions() -> Result<(), Box<dyn Error + Send + Sync>> {
```
Comment (rylev) [inline sdk/storage/tests/data_lake.rs:110]: Can this not be run as a mock testing framework test?

**Review by rylev — COMMENTED** [review]
Looks good to me. I'd prefer if we could have some mock testing framework tests that verify that this works though

File: sdk/core/src/request_options/content_length.rs (hunk 1 above)
Comment (thovoll) [reply sdk/core/src/request_options/content_length.rs:8]: Yes, just because it's obvious. Open to leaving it in though it provides some value I overlooked.

File: sdk/core/src/request_options/rename_source.rs (hunk 2 above)
Comment (thovoll) [reply sdk/core/src/request_options/rename_source.rs:6]: Good point, let me check, but probably only used in data_lake. Will add a comment as well.

File: sdk/storage/examples/data_lake_00.rs (hunk 3 above)
Comment (thovoll) [reply sdk/storage/examples/data_lake_00.rs:73]: I moved the file specific operations to the data_lake_01.rs example

**Comment by thovoll** [comment]
@rylev I addressed your feedback, this should be ready to go!

File: sdk/core/src/request_options/content_length.rs (hunk 1 above)
Comment (rylev) [reply sdk/core/src/request_options/content_length.rs:8]: I'd personally like if every public function and type were documented even if obvious.

**Review by rylev — APPROVED** [review]
🎉

File: sdk/core/src/request_options/content_length.rs (hunk 1 above)
Comment (thovoll) [reply sdk/core/src/request_options/content_length.rs:8]: I see the value of consistent public docs and might have gone a bit too far removing this, probably because I just encountered [these `PathRenameMode` docs](https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/create#pathrenamemode) during this PR - I still don't know what this parameter does and just copied what the .Net SDK does ;)

However, currently only 1 of the other 30 types (`MaxItemCount`) has such a comment. Maybe we need an issue to add docs to all public functions (maybe per crate)?
